message_formatting/dev.py

```diff
@@ -6,6 +6,10 @@
 async def my_command(ctx: discord.ApplicationContext, arg1, arg2) -> None:
     # Do stuff
     await ctx.respond("Done!", view=feedback(ctx.author, "my_command" | ctx.command.name))
+
+Note from @GabeMillikan:
+    Once we're done testing this (amazing) feature, the file should be renamed to
+    `message_formatting/views/feedback.py` and should use the production database
```

Comment:
Disagreed. This should stay in its own database. It is a feature for developers and we intent to give this feedback to more people that should not have access to the production database.

I am aware that we can "just select * from feedback". But this is, and will remain a _development_ feature. The database will also be flushed after we've received feedback and decided if a feature is good or not. Which is not something you do to production databases.

I also have my doubts about the new suggested location. But that is up for discussion. Merging with production is not.

Reply:
I see, I misunderstood your intended purpose for this file.

> we intent to give this feedback to more people that should not have access to the production database

I'll agree that this is a valid reason to keep the data in a separate database, we can easily adjust `/view-database` to serve different databases to different audiences.

> The database will also be flushed

I honestly don't see a reason to do this. It seems like it could be nice to have historical records of good and bad features and the ability to graph a feature's public opinion over time (long term). If we're only interested in ratings from the last month (or whatever) then we can simply do `select * from ratings where datetime > 1 month ago`.

> I also have my doubts about the new suggested location.

I would prefer to organize files based on their _abilities_ (or what they contain) rather than their _purpose_ (what those abilities are used for), if that makes sense. This file has the _ability_ to add views to messages whose  _purpose_ is to gather feedback for developers on new features.

Naming files based on the _purpose_ unnecessarily places limits the end-user (whatever is importing the file). This file specifically will never be used for any other purpose, but I like to follow general naming conventions with as few exceptions as possible.

Reply:
With flush I meant to put in a clean database for said intended feature we are testing at the time. Historical data will never be fully deleted and kept in an archive. Just not in the same db.

I also get where your naming convention comes from. I tend to name files for their purpose, but we can do it your way if you want to.

Reply:
> With flush I meant to put in a clean database for said intended feature we are testing at the time. Historical data will never be fully deleted and kept in an archive. Just not in the same db.

👍 

> I tend to name files for their purpose, but we can do it your way if you want to.

How about we do a vote among pax contributors? We both have our reasons to have a certain preference & neither way is objectively correct so we could just go with the most popular option.

Reply:
I mean, most files and locations are already named by your naming convention. Changing everything back will just be a pain in the rear, so I guess I'll just have to get used to it..

(Don't blame me if I misname a file in the near future though)

Reply:
Sounds good :)